docs/resources/blocks-and-transactions.md

```diff
@@ -0,0 +1,1560 @@
+
+[Top](#table-of-content)
+
+## Common Transaction Details
+
+### 1. Block Height
+
+- [Tendermint Block API](#_1-tendermint-block-api): `result.block.header.height`
+- [Tendermint Block Results API](#_2-blocktendermint-block-results-api): `result.height`
+- [Cosmos Transaction Query API](#_3-cosmos-transaction-query-api): `tx_response.height`
+- [Cosmos Transaction Search API](#_4-cosmos-transaction-search-api): `tx_response[index].height`
+
+### 2. Transaction Hash
+
+- [Tendermint Block API](#_1-tendermint-block-api): `Uppercase(SHA256(Base64Decode(result.block.data.txs[index])))`
+- [Tendermint Block Results API](#_2-blocktendermint-block-results-api): Not available, should use [Tendermint Block API](#tendermint-block-api). Match transaction `[index]` in `result.txs_results` with `result.block.data.txs[index]`
+- [Cosmos Transaction Query API](#_3-cosmos-transaction-query-api): `tx_response.txhash`
+- [Cosmos Transaction Search API](#_4-cosmos-transaction-search-api): `tx_response[index].txhash`
+
+### 3. Transaction Fee
+
+- [Cosmos Transaction Query API](#_3-cosmos-transaction-query-api): `tx.auth_info.fee.amount`
+- [Cosmos Transaction Search API](#_4-cosmos-transaction-search-api): `tx[index].auth_info.fee.amount`
+
+Transaction fee is an [Asset Array](#_2-array), meaning that a transaction can pay fee in more than one token types.
+
+### 4. Assets and Amount
+
+There are mainly two types of assets and amount representation:
+
+#### 1. Single object
+
+This is commonly seen in `staking` module but may appear in other module as well. It represents a single token type.
+
+**Example**
+
+```json
+{
+  "denom": "basecro",
+  "amount": "10000"
+}
+```
+
+where `denom` is the asset type and `amount` is the amount and `basecro` is the basic unit of CRO token (where 10^8 `basecro` = 1 cro)
+
+Note that the `amount` is always in string for precision accuracy. Please make sure your language is capable to handle big integer number.It is highly recommended to use library similar to [bignumber.js](https://mikemcl.github.io/bignumber.js/) in your language to handle the `amount`.
+
+<!-- omit in toc -->
+
+
+
+#### 2. Array
+
+This is commonly seen in most message types. It represents a list of tokens.
+
+At the time of writing there will only be a single entry in this array because `basecro` (or `basetcro` in Croeseid Testnet) is the only supported asset on Crypto.org Chain. However, after IBC transfer and other coins issuance methods are enabled, there will be more asset types, the coin source tracing and their denomination can be found in [here](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-001-coin-source-tracing.md)
```

Comment:
```suggestion 
At the time of writing there will only be a single entry in this array because `basecro` (or `basetcro` in Croeseid Testnet) is the only supported asset on Crypto.org Chain. However, after IBC transfer and other coins issuance methods are enabled, there will be more asset types, the coin source tracing and their denomination can be found [here](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-001-coin-source-tracing.md)
```